### DISCUSSION TITLE
Prisma Delete function: TypeError: Cannot read properties of undefined (reading 'model')

### DISCUSSION THREAD

**Discussion by shedp** [prisma/prisma#23244] [Q&A]
### Question

I am using NextJS client with Prisma (mongoDB). I am trying to create a delete function:

deleteEvent.ts Error deleting event: TypeError: Cannot read properties of undefined (reading 'events')

My other prisma.events.findMany and prisma.user.findUnique functions are working perfectly fine which suggests that the Prisma client is able to interact with the models.

I can't see any syntax errors and the user does not need to be authenticated (as that is done on the frontend).

Can anyone help?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model Events {
  id String @id @default(auto()) @map("_id") @db.ObjectId
  userId String @db.ObjectId
  title String
  description String
  category String
  date DateTime
  repeatable Boolean @default(false)
  frequency Int?
  period String?
  createdAt DateTime  @default(now())

  user User @relation(fields: [userId], references: [id], onDelete: Cascade)
}
```

```ts
import prisma from "@/app/libs/prismadb";

export default async function deleteEvent(eventId: string) {
  try {
console.log(eventID);
    const deletedEvent = await prisma.events.delete({
      where: {
        id: eventId,
      },
    });
    console.log("Deleted Event:", deletedEvent);
    return deletedEvent;
  } catch (error: any) {
    console.error("Error deleting event:", error);
    throw new Error(error);
  }
}
```
```
const onDelete = async (deleteId: string) => {
    try {
      await deleteEvent(deleteId);
      toast.success("Event deleted!");
    } catch (error) {
      toast.error("Something went wrong");
    }
  };
...
return <Button type="delete" onClick={() => onDelete(id)} />
```

- Database: MongoDB
- Node.js version: v20.9.0
- Prisma version: 5.7.1

**Comment by nurul3101** [comment]
Hello @shedp 👋 

Can you share how you instantiate your `PrismaClient` in `"@/app/libs/prismadb`?

The error message is likely because the value of prisma is undefined while evaluating this line:
```ts
const deletedEvent = await prisma.events.delete({
```

On a separate note, In your deleteEvent function, you logged `console.log(eventID); `but your function parameter is `eventId`, which might be a Typo.

**Reply by shedp** [reply]
Hi @nurul3101,

Thanks for replying, my @/app/libs/prismadb looks like the following:

`import { PrismaClient } from "@prisma/client";

declare global {
  var prisma: PrismaClient | undefined;
}

let prisma: PrismaClient;

if (typeof window === "undefined") {
  if (process.env.NODE_ENV === "production") {
    prisma = new PrismaClient();
  } else {
    if (!global.prisma) {
      global.prisma = new PrismaClient();
    }
    prisma = global.prisma;
  }
}

export default prisma;`

**Reply by ludralph** [reply]
Hi @shedp 

> In your deleteEvent function, you logged console.log(eventID); but your function parameter is eventId, which might be a Typo.

Did you modify this line as suggested my my colleague Nurul? Also, are you still running into this issue?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
